### ISSUE TITLE
In shard mode, close throw error "CROSSSLOT Keys in request don't hash to the same slot"

### ISSUE THREAD

**Issue by ngot** [socketio/socket.io-redis-adapter#500]
I had a chance to try the new shard PubSub adapter. Mostly works good, but when enable `cleanupEmptyChildNamespaces`, the `close` function will throw "CROSSSLOT Keys in request don't hash to the same slot". The root cause is https://github.com/redis/node-redis/issues/2502 

Currently,

```js
    close() {
        const channels = [this.channel, this.responseChannel];
        if (this.opts.subscriptionMode === "dynamic") {
            this.rooms.forEach((_sids, room) => {
                const isPublicRoom = !this.sids.has(room);
                if (isPublicRoom) {
                    channels.push(this.dynamicChannel(room));
                }
            });
        }
       // channels has at least 2 different strings, `sUnsubscribe` assumes 
       // they would be hashed to the same slot which is not correct.
       return this.subClient.sUnsubscribe(channels);
    }
```

A potential workaround 

```js
return Promise.all(channels.map((channel) => this.subClient.sUnsubscribe(channel)));
// return this.subClient.sUnsubscribe(channels);

```

**Comment by darrachequesne** [comment]
Hi! Thanks for raising the issue, I'm digging into this.

**Comment by ngot** [comment]
According to https://github.com/redis/node-redis/issues/2502#issuecomment-1541523864, the `sUnsubscribe(Array[])` assumes all the channels use ["hash tags"](https://redis.io/docs/reference/cluster-spec/#hash-tags), so they are allocated to the same slot. 

In socket.io-redis-adapter, there is dynamic channels, so `sUnsubscribe` one by one looks more doable.

**Comment by darrachequesne** [comment]
This should be fixed by https://github.com/socketio/socket.io-redis-adapter/commit/2da8d9e57afbed3201f818bca77ac17ce9636fa3, included in version [8.2.1](https://github.com/socketio/socket.io-redis-adapter/releases/tag/8.2.1).

@ngot thanks for the analysis :+1:
